cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/co/cask/cdap/etl/proto/v2/ETLConfig.java

```diff
@@ -47,9 +47,10 @@
   private final co.cask.cdap.etl.proto.v1.ETLStage source;
   private final List<co.cask.cdap.etl.proto.v1.ETLStage> sinks;
   private final List<co.cask.cdap.etl.proto.v1.ETLStage> transforms;
+  private final int numOfRecordsPreview;
```

Comment:
Since this class is instantiated through gson, its better to make this an Integer. That way, you can tell when the user didn't specify anything, and default it here, instead of treating 0 as a special number and defaulting later.